crates/marketplace/src/service.rs

```diff
@@ -375,6 +378,13 @@ where
             .try_collect()
             .await
     }
+
+    async fn txn_status(
+        &self,
+        txn_hash: Commitment<<Types as NodeType>::Transaction>,
+    ) -> Result<TransactionStatus, BuildError> {
+        self.read(|state| state.txn_status(txn_hash)).await
```

Comment:
Sorry, isn't this just endless recursion? It looks to me like there's no implementation for marketplace in this version of the PR